### PR TITLE
Create a Python App that Extracts images, Scan For text Info, Transcribe and save to a .txt file

### LINKED ISSUE #6897: Create a Python App that Extracts images, Scan For text Info, Transcribe and save to a .txt file
#### Proposed article introduction

Python is a huge and popular language that many developers enjoy working with. Its human-readable syntax makes it a popular language and an easy to learn path. As a developer, you might need to extract information from an image. Images contain visual text that you can extract and add to the relevant text document. Python makes a good choice to create a program that extracts such text from any given image. In this guide, we will learn how to create s python app using the tesseract. We will create and write a Python script that extracts images, scan text, transcribe and save it to a .txt file.

#### Key takeaways 

- Why create a Python App that Extracts images info
- What is Python tesseract
- Set up a Python tesseract project
- Install dependencies
- Create a Python tesseract script that Extracts images info
- Test the Python tesseract app

#### Article quality

The overall guide aims to help the reader build an app that Extracts images info using Python scrips.

#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by Wachira48**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [ ] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [ ] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [ ] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [ ] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [ ] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [ ] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [ ] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [ ] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [ ] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [ ] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [ ] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [ ] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [ ] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [ ] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #6897 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

File: content/articles/python-app-that-extracts-images-scan-text-transcribe-save-to-txt-file/index.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
+Python is a huge and popular language that many developers enjoy working with. Its human-readable syntax makes it a popular language and an easy to learn path. As a developer, you might need to extract information from an image. Images contain visual text that you can extract and add to the relevant text document. Python makes a good choice to create a program that extracts such text from any given image. In this guide, we will learn how to create s python app using the tesseract. We will create and write a Python script that extracts images, scan text, transcribe and save it to a .txt file.
```
Comment (srishilesh) [inline content/articles/python-app-that-extracts-images-scan-text-transcribe-save-to-txt-file/index.md:1]: Break down this intro. Check other articles

File: content/articles/python-app-that-extracts-images-scan-text-transcribe-save-to-txt-file/index.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
```
Comment (srishilesh) [inline content/articles/python-app-that-extracts-images-scan-text-transcribe-save-to-txt-file/index.md:2]: Table of contents & prerequisities

File: content/articles/python-app-that-extracts-images-scan-text-transcribe-save-to-txt-file/index.md [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+### Set up tesseract OCR
```
Comment (srishilesh) [inline content/articles/python-app-that-extracts-images-scan-text-transcribe-save-to-txt-file/index.md:3]: Add topic on Tesseract

File: content/articles/python-app-that-extracts-images-scan-text-transcribe-save-to-txt-file/index.md [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+
+To extract text from images, Python uses OCR technology. OCR (Optical Character Recognition) is a technology that is used to recognize text in images. It is used to convert tight handwritten or printed text into machine-readable text. OCR engine detects text in images and overlays the text onto PDFs.
+
+To use OCR, you need to [install and configure](https://medium.com/quantrium-tech/installing-and-using-tesseract-4-on-windows-10-4f7930313f82) tesseract on your computer. First, download the Tesseract OCR executables [here](https://codetoprosper.com/tesseract-ocr-for-windows/). While installing this executable, make sure you copy the tesseract installation path and it to your [system environment varibales](https://codetoprosper.com/tesseract-ocr-for-windows/). Once the process is done, run the `tesseract -v` command to verify the OCR is installed.
+
+![tesseract](tesseract.jpg)
```
Comment (srishilesh) [inline content/articles/python-app-that-extracts-images-scan-text-transcribe-save-to-txt-file/index.md:9]: Incorrect path

File: content/articles/python-app-that-extracts-images-scan-text-transcribe-save-to-txt-file/index.md [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+    # Print input
+    print(Fore.GREEN + "[!] Insert path to PDF file:" + Fore.RESET)
+    inputUser = input()
+
+    # Print an alert if input is not valid, if not, call to fun reDoc
+    if(inputUser == "" or len(inputUser.split("\\")) == 1):
+        print(Fore.RED + "[X] Please put a valid PATH to a file" + Fore.RESET)
+    else:
+        extIm(inputUser)
+```
```
Comment (srishilesh) [inline content/articles/python-app-that-extracts-images-scan-text-transcribe-save-to-txt-file/index.md:118]: Code explanation - state objectives of this code.

Put them as bulleted points

File: content/articles/python-app-that-extracts-images-scan-text-transcribe-save-to-txt-file/index.md [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+                image_bytes = base_image["image"]
+                # get the image extension
+                image_ext = base_image["ext"]
+                # load it to PIL
+                image = Image.open(io.BytesIO(image_bytes))
+                # save it to local disk
+                image.save(
+                    open(f"images/image{page_index+1}_{image_index}.{image_ext}", "wb"))
+    reImg()
+```
```
Comment (srishilesh) [inline content/articles/python-app-that-extracts-images-scan-text-transcribe-save-to-txt-file/index.md:175]: Modularize the code and explain each method

File: content/articles/python-app-that-extracts-images-scan-text-transcribe-save-to-txt-file/index.md [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+        # Showing img input
+        cv2.imshow('Image', image)
+        # 0.5 milisecond
+        cv2.waitKey(1000)
+
+    # Create and write file txtResult.txt
+    print(Fore.CYAN + "[.] Writing txtResult.txt" + Fore.RESET)
+    fileTxt = open(f"{dirName[1]}/txtResult.txt", "w")
+    fileTxt.write(textScanned)
+    print(Fore.GREEN + "[!] File Writted" + Fore.RESET)
```
Comment (srishilesh) [inline content/articles/python-app-that-extracts-images-scan-text-transcribe-save-to-txt-file/index.md:212]: Break it down

File: content/articles/python-app-that-extracts-images-scan-text-transcribe-save-to-txt-file/index.md [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+```
+
+```py
+# Call to fun main
+gInUs()
+```
+
```
Comment (srishilesh) [inline content/articles/python-app-that-extracts-images-scan-text-transcribe-save-to-txt-file/index.md:219]: This is an abrupt ending

File: content/articles/python-app-that-extracts-images-scan-text-transcribe-save-to-txt-file/index.md [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+### Test the Python tesseract app
+
+![access-denied](access-denied.jpg)
+
+![elevation-error](elevation-error.jpg)
```
Comment (srishilesh) [inline content/articles/python-app-that-extracts-images-scan-text-transcribe-save-to-txt-file/index.md:224]: No conclusion or further reading section.

**Review by srishilesh — CHANGES_REQUESTED** [review]
Hey @Wachira48 ,

I believe this is your 4th article at Section EngEd.

Before you start with this article, did you go through the reviews of the 3 articles that you had published before?

I find many instances where the suggestions/revisions were not applied:
- Image path is incorrect
- No explanation of code snippets
- Many missing topics.
- And, a few more.

We highly recommend you to create PRs that are fully publishable/no revisions required. If there are any glancing errors, we will be closing the PR.

Hence closing this PR.

Thanks.

**Comment by srishilesh** [comment]
Hey @Wachira48 ,

Just checking if we have got any updates on this PR.

Let me know if you need any help.

**Comment by Wachira48** [comment]
Yes, @srishilesh. I have them ready

**Comment by srishilesh** [comment]
@Wachira48 You can push your changes now

**Comment by Wachira48** [comment]
I have added the changes @srishilesh

**Review by srishilesh — APPROVED** [review]
Hey @Wachira48 ,

Thanks for making the changes.

Moving for the final review.

Thanks.

**Comment by Kayere** [comment]
Great work gentlemen 🚀.

**Pull request merged**